src/Generator/Program.fs

```diff
@@ -0,0 +1,390 @@
+﻿// Learn more about F// at http://fsharp.org
+open Types
+open SpecificationOpenGL
+open Util
+open Constants
+open System.IO
+open CommandLine
+
+type options =
+    { [<Option('i', "input", Required = true,
+               HelpText = "Path to specification file.")>]
+      pathToSpecificationFile: string
+      [<Option('o', "output", Required = true,
+               HelpText = "Path to output directory.")>]
+      pathToOutputDirectory: string }
+
+open Formatting
+
+let autoGenerateAdditionalOverloadForType (func: PrintReadyTypedFunctionDeclaration) =
+    let lengthParamsSet =
+        func.parameters
+        |> Array.choose(fun p -> p.lengthParamName)
+        |> Set.ofArray
+
+    let rec unwrapTyFromPointer ty =
+        match ty with
+        | Pointer ty -> unwrapTyFromPointer ty
+        | _ -> ty
+    let transformPointerTy i transform ty =
+        let transformPointerTy ty =
+            match ty with
+            | Pointer(inner) ->
+                let flattenedTy = unwrapTyFromPointer inner
+                let res = flattenedTy |> transform
```

Comment:
Let's make it more clear that the output of transform is a bool

Reply:
`transform` has the signature `GLType ->`GLType`

Reply:
I think I've misread your code. 
In this case, I vote making clear that it's a OpenGL type :)